packages/host/tests/acceptance/interact-submode-test.gts

```diff
@@ -980,5 +981,183 @@ module('Acceptance | interact submode tests', function (hooks) {
         'select is preserved',
       );
     });
+
+    test('containsMany string field preserves focus while typing', async function (assert) {
+      const receivedEventDeferred = new Deferred<void>();
+      const messageService = getService('message-service');
+      const typedText = 'Ada';
+      const inputSelector =
+        '[data-test-contains-many="names"] [data-test-item="0"] input';
+
+      messageService.listenerCallbacks.get(testRealmURL)!.push((e) => {
+        if (
+          e.eventName === 'index' &&
+          e.indexType === 'incremental-index-initiation'
+        ) {
+          return; // ignore the index initiation event
+        }
+        receivedEventDeferred.fulfill();
+      });
+
+      await visitOperatorMode({
+        stacks: [
+          [
+            {
+              id: `${testRealmURL}FocusTest/1`,
+              format: 'edit',
+            },
+          ],
+        ],
+      });
+
+      await click('[data-test-contains-many="names"] [data-test-add-new]');
+      let inputElement = find(inputSelector) as HTMLInputElement;
+
+      let focusStates: boolean[] = [];
+      let inputEventCount = 0;
+      const handleInputEvent = (event: Event) => {
+        let target = event.target;
+        if (!(target instanceof HTMLInputElement)) {
+          return;
+        }
+        if (!target.matches(inputSelector)) {
+          return;
+        }
+        inputEventCount += 1;
+        focusStates.push(document.activeElement === target);
+      };
+
+      document.addEventListener('input', handleInputEvent, true);
+      try {
+        inputElement.focus();
+        await typeIn(inputSelector, typedText);
+      } finally {
+        document.removeEventListener('input', handleInputEvent, true);
+      }
+
+      await receivedEventDeferred.promise;
+      await settled();
+
+      assert.strictEqual(
+        inputEventCount,
+        typedText.length,
+        'input events are triggered for each keystroke',
+      );
+      assert.true(
+        focusStates.every(Boolean),
+        'focus is preserved on the input element during typing',
+      );
+      inputElement = find(inputSelector) as HTMLInputElement;
+      assert.strictEqual(
+        document.activeElement,
+        inputElement,
+        'focus is preserved on the input element after typing',
+      );
+      assert.dom(inputSelector).hasValue(typedText);
+    });
+
+    test('containsMany field def preserves focus while typing', async function (assert) {
```

Comment:
chris really wanted to make sure this pattern works too, so I added a test for it as well